### PR TITLE
feat: support friend statistics

### LINKED ISSUE #160: Support new Messaging API endpoints for friend statistics added on 07/08
Support new API announced on 20190708:
https://developers.line.biz/en/news/#new-messaging-api-endpoints-for-friend-statistics-2019-07-08

- [ ] [Get number of message deliveries](https://developers.line.biz/en/reference/messaging-api/#get-number-of-delivery-messages)
- [ ] [Get number of followers](https://developers.line.biz/en/reference/messaging-api/#get-number-of-followers)
- [ ] [Get friend demographics](https://developers.line.biz/en/reference/messaging-api/#get-demographic)

### PR DESCRIPTION
**Pull request by xingoxu**
fix #160 

- [x] [Get number of message deliveries](https://developers.line.biz/en/reference/messaging-api/#get-number-of-delivery-messages)
- [x] [Get number of followers](https://developers.line.biz/en/reference/messaging-api/#get-number-of-followers)
- [x] [Get friend demographics](https://developers.line.biz/en/reference/messaging-api/#get-demographic)

### CODE REVIEW

File: lib/types.ts [hunk 1]
```diff
@@ -1801,6 +1801,144 @@ export type NumberOfSentBroadcastMessages = {
...
+
+export type NumberOfMessageDeliveriesResponse =
+  | InsightStatisticsResponse
+  | NumberOfMessageDeliveries;
+
+export type NumberOfFollowersResponse =
+  | InsightStatisticsResponse
+  | NumberOfFollowers;
+
+type percentageAble = {
```
Comment (SombreroElGringo) [inline lib/types.ts]: Pascal Case must be used for the type's name

**Comment by xingoxu** [comment]
@SombreroElGringo Thank you!

File: lib/types.ts [hunk 2]
```diff
@@ -1801,6 +1793,146 @@ export type NumberOfSentBroadcastMessages = {
...
+  available: boolean;
+  /**
+   * Percentage per gender
+   */
+  genders?: Array<
+    {
+      /**
+       * Gender
+       */
+      gender: "unknown" | "male" | "female";
```
Comment (kawasako) [inline lib/types.ts]: Is there any possibility that other strings will be returned from the API..?

**Comment by xingoxu** [comment]
@kawasako
At this time, it won't be other value.

2019年7月30日(火) 16:28 Kawasaki Kohei <notifications@github.com>:

**Comment by kawasako** [comment]
> At this time, it won't be other value.

I see, thanks. LTGM

**Pull request merged**